### PR TITLE
Closes #249

### LINKED ISSUE #249: Matrix-Only BasicStatistics fail
--------
Issue Description
--------
##### What did you expect to see happen?
Run basic statistics with `only sensitivity` and no single-valued results

##### What did you see instead?
Error:
```
Traceback (most recent call last):
  File "/Users/talbpw/projects/raven/framework/Driver.py", line 278, in <module>
    raven()
  File "/Users/talbpw/projects/raven/framework/Driver.py", line 231, in raven
    simulation.run()
  File "/Users/talbpw/projects/raven/framework/Simulation.py", line 1051, in run
    stepInstance.takeAstep(stepInputDict)
  File "/Users/talbpw/projects/raven/framework/Steps.py", line 238, in takeAstep
    self._localTakeAstepRun(inDictionary)
  File "/Users/talbpw/projects/raven/framework/Steps.py", line 384, in _localTakeAstepRun
    model.collectOutput(finishedJob,output)
  File "/Users/talbpw/projects/raven/framework/Models/PostProcessor.py", line 213, in collectOutput
    self.interface.collectOutput(finishedjob,output)
  File "/Users/talbpw/projects/raven/framework/PostProcessors/BasicStatistics.py", line 362, in collectOutput
    self._writeText(output,outputDictionary,methodToTest,separator)
  File "/Users/talbpw/projects/raven/framework/PostProcessors/BasicStatistics.py", line 434, in _writeText
    output.write(separator.join(valueStrFormat(v) for v in [' ']+vecTargets)+os.linesep)
  File "/Users/talbpw/projects/raven/framework/PostProcessors/BasicStatistics.py", line 434, in <genexpr>
    output.write(separator.join(valueStrFormat(v) for v in [' ']+vecTargets)+os.linesep)
NameError: free variable 'valueStrFormat' referenced before assignment in enclosing scope
```

##### Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.
See Workshop, Forward Sampling, example 6, when you remove all EXCEPT for the last sensitivity block.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or improvement? ***defect***
- [x] 2. Is it tagged with a priority: critical, normal or minor? ***critical***
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements? ***n/a***
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users. ***no. just a crash***
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.) ***yes***

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed? *Yes*
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.) *Yes*
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)? *Cannot produce wrong results since results in crash, so email would be good but not required.*
- [x] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)? *No stable branch currently*
- [x] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided? *Will be closed with merge request*

Close reviewed by @joshua-cogliati-inl

### PR DESCRIPTION
**Pull request by alfoa**
--------
Pull Request Description
--------
##### What issue does this change request address?
Closes #249
##### What are the significant changes in functionality due to this change request?
This addresses issue #249 

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code. *Done*
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts). *No changes*
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details. *Only moving code*
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass. *they pass*
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True. *Test added*
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync. *No change*
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done. *Close checklist done*
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added? *No change*

Checklist will be done when automated tests pass.

### CODE REVIEW

File: tests/framework/PostProcessors/BasicStatistics/tests [hunk 1]
```diff
@@ -83,6 +83,14 @@
   UnorderedXml = 'basicStatsSensitivity/stats.xml'
   rel_err = 0.0000001
  [../]
+ [./sensitivityOnly]
+  type = 'RavenFramework'
+  input = 'sensitivity_only.xml'
```
Comment (joshua-cogliati-inl) [inline tests/framework/PostProcessors/BasicStatistics/tests]: sensitivity_only.xml has not been checked in.

Comment (alfoa) [inline tests/framework/PostProcessors/BasicStatistics/tests]: added

File: tests/framework/PostProcessors/BasicStatistics/sensitivity_only.xml [hunk 2]
```diff
@@ -0,0 +1,118 @@
+<?xml version="1.0" ?>
+<Simulation verbosity="silent">
+  <RunInfo>
+    <WorkingDir>basicStatsSensitivityOnly</WorkingDir>
+    <batchSize>1</batchSize>
+    <Sequence>sample,PP</Sequence>
+  </RunInfo>
+  <TestInfo>
+    <name>framework/PostProcessors/BasicStatistics.sensitivityOnly</name>
+    <author>wangc</author>
```
Comment (joshua-cogliati-inl) [inline tests/framework/PostProcessors/BasicStatistics/sensitivity_only.xml]: Hm, author and created should probably be updated.

File: tests/framework/PostProcessors/BasicStatistics/tests (hunk 1 above)
Comment (joshua-cogliati-inl) [inline tests/framework/PostProcessors/BasicStatistics/tests]: Okay, thanks for fixing.

File: tests/framework/PostProcessors/BasicStatistics/sensitivity_only.xml (hunk 2 above)
Comment (alfoa) [inline tests/framework/PostProcessors/BasicStatistics/sensitivity_only.xml]: right

**Comment by joshua-cogliati-inl** [comment]
Hm, mac failed with:
```
framework/ROM/SKLearn.linearMultiTaskElasticNet FAILED (CWD /Users/raven_test/civet/client_root_0/raven/tests/framework/ROM/Sobol METHOD opt Expected files not created /Users/raven_test/civet/client_root_0/raven/tests/framework/ROM/SKLearn/data/outLinearMultiTaskElasticNet.xml)
framework/ROM/SKLearn.linearMultiTaskElasticNet: Working Directory: /Users/raven_test/civet/client_root_0/raven/tests/framework/ROM/SKLearn
framework/ROM/SKLearn.linearMultiTaskElasticNet: Running command: python /Users/raven_test/civet/client_root_0/raven/framework/Driver.py linearMultiTaskElasticNet.xml
framework/ROM/SKLearn.linearMultiTaskElasticNet: Traceback (most recent call last):
framework/ROM/SKLearn.linearMultiTaskElasticNet:   File "/Users/raven_test/civet/client_root_0/raven/framework/Driver.py", line 48, in <module>
framework/ROM/SKLearn.linearMultiTaskElasticNet:     from Simulation import Simulation
framework/ROM/SKLearn.linearMultiTaskElasticNet:   File "/Users/raven_test/civet/client_root_0/raven/framework/Simulation.py", line 36, in <module>
framework/ROM/SKLearn.linearMultiTaskElasticNet:     import Steps
framework/ROM/SKLearn.linearMultiTaskElasticNet:   File "/Users/raven_test/civet/client_root_0/raven/framework/Steps.py", line 43, in <module>
framework/ROM/SKLearn.linearMultiTaskElasticNet:     import Files
framework/ROM/SKLearn.linearMultiTaskElasticNet:   File "/Users/raven_test/civet/client_root_0/raven/framework/Files.py", line 34, in <module>
framework/ROM/SKLearn.linearMultiTaskElasticNet:     from utils import utils, mathUtils
framework/ROM/SKLearn.linearMultiTaskElasticNet:   File "/Users/raven_test/civet/client_root_0/raven/framework/utils/mathUtils.py", line 27, in <module>
framework/ROM/SKLearn.linearMultiTaskElasticNet:     from scipy import interpolate, stats, integrate
framework/ROM/SKLearn.linearMultiTaskElasticNet:   File "/opt/raven_libs/envs/raven_libraries/lib/python2.7/site-packages/scipy/interpolate/__init__.py", line 158, in <module>
framework/ROM/SKLearn.linearMultiTaskElasticNet:     from .interpolate import *
framework/ROM/SKLearn.linearMultiTaskElasticNet:   File "/opt/raven_libs/envs/raven_libraries/lib/python2.7/site-packages/scipy/interpolate/interpolate.py", line 27, in <module>
framework/ROM/SKLearn.linearMultiTaskElasticNet:     from .interpnd import _ndim_coords_from_arrays
framework/ROM/SKLearn.linearMultiTaskElasticNet:   File "scipy/interpolate/interpnd.pyx", line 1, in init scipy.interpolate.interpnd (scipy/interpolate/interpnd.c:25150)
framework/ROM/SKLearn.linearMultiTaskElasticNet:   File "/opt/raven_libs/envs/raven_libraries/lib/python2.7/site-packages/scipy/spatial/__init__.py", line 92, in <module>
framework/ROM/SKLearn.linearMultiTaskElasticNet:     from .ckdtree import *
framework/ROM/SKLearn.linearMultiTaskElasticNet:   File "scipy/spatial/ckdtree.pyx", line 29, in init scipy.spatial.ckdtree (scipy/spatial/ckdtree.cxx:17696)
framework/ROM/SKLearn.linearMultiTaskElasticNet:   File "/opt/raven_libs/envs/raven_libraries/lib/python2.7/multiprocessing/__init__.py", line 136, in cpu_count
framework/ROM/SKLearn.linearMultiTaskElasticNet:     raise NotImplementedError('cannot determine number of cpus')
framework/ROM/SKLearn.linearMultiTaskElasticNet: NotImplementedError: cannot determine number of cpus
framework/ROM/SKLearn.linearMultiTaskElasticNet: 
framework/ROM/SKLearn.linearMultiTaskElasticNet FAILED (CWD /Users/raven_test/civet/client_root_0/raven/tests/framework/ROM/Sobol METHOD opt Expected files not created /Users/raven_test/civet/client_root_0/raven/tests/framework/ROM/SKLearn/data/outLinearMultiTaskElasticNet.xml) (reprint)
```

This might be an intermittent.

**Comment by alfoa** [comment]
@joshua-cogliati-inl approved this.

Merging...

**Pull request merged**